addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueDevice.java

```diff
@@ -352,10 +352,35 @@ private void setState(State itemState) {
             successApplied.put("effect", newState.effect); // Pretend that worked
         }
         if (newState.xy != null) {
-            errorApplied.add("xy");
+            try {
+                HueStateColorBulb c = as(HueStateColorBulb.class);
+                c.colormode = ColorMode.xy;
+                c.bri = as(HueStateBulb.class).bri;
+                c.xy[0] = newState.xy.get(0);
+                c.xy[1] = newState.xy.get(1);
+                command = c.toHSBType();
+                successApplied.put("xy", newState.xy);
+            } catch (ClassCastException e) {
```

Comment:
What part can throw this?

Reply:
as(HueStateColorBulb.class) in case someone tries to set a color on a "only dimmable" lamp